cpp/tests/spatial/linestring_polygon_distance_test.cpp

```diff
@@ -0,0 +1,230 @@
+  rmm::cuda_stream_view stream() { return cudf::get_default_stream(); }
+  void SetUp()
+  {
+    collection_type_id _;
+    std::tie(_, empty_linestring_column)      = make_linestring_column<T>({0}, {}, stream());
+    std::tie(_, empty_multilinestring_column) = make_linestring_column<T>({0}, {0}, {}, stream());
+    std::tie(_, empty_polygon_column)         = make_polygon_column<T>({0}, {0}, {}, stream());
+    std::tie(_, empty_multipolygon_column)    = make_polygon_column<T>({0}, {0}, {0}, {}, stream());
+  }
+
+  geometry_column_view empty_linestring()
+  {
+    return geometry_column_view(
+      empty_linestring_column->view(), collection_type_id::SINGLE, geometry_type_id::LINESTRING);
+  }
+
+  geometry_column_view empty_multilinestring()
+  {
+    return geometry_column_view(empty_multilinestring_column->view(),
+                                collection_type_id::MULTI,
+                                geometry_type_id::LINESTRING);
+  }
+
+  geometry_column_view empty_polygon()
+  {
+    return geometry_column_view(
+      empty_polygon_column->view(), collection_type_id::SINGLE, geometry_type_id::POLYGON);
+  }
+
+  geometry_column_view empty_multipolygon()
+  {
+    return geometry_column_view(
+      empty_multipolygon_column->view(), collection_type_id::MULTI, geometry_type_id::POLYGON);
+  }
+
+  void run_single(geometry_column_view linestrings,
+                  geometry_column_view polygons,
+                  std::initializer_list<T> expected)
+  {
+    auto got = pairwise_linestring_polygon_distance(linestrings, polygons);
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(*got, fixed_width_column_wrapper<T>(expected));
+  }
+
+  std::unique_ptr<cudf::column> empty_linestring_column;
+  std::unique_ptr<cudf::column> empty_multilinestring_column;
+  std::unique_ptr<cudf::column> empty_polygon_column;
+  std::unique_ptr<cudf::column> empty_multipolygon_column;
+};
+
+struct PairwiseLinestringPolygonDistanceTestUntyped : testing::Test {
+  rmm::cuda_stream_view stream() { return cudf::get_default_stream(); }
+};
+
+using TestTypes = ::testing::Types<float, double>;
+
+TYPED_TEST_CASE(PairwiseLinestringPolygonDistanceTest, TestTypes);
+
+TYPED_TEST(PairwiseLinestringPolygonDistanceTest, SingleToSingleEmpty)
+{
+  CUSPATIAL_RUN_TEST(this->run_single, this->empty_linestring(), this->empty_polygon(), {});
+};
+
+TYPED_TEST(PairwiseLinestringPolygonDistanceTest, SingleToMultiEmpty)
+{
+  CUSPATIAL_RUN_TEST(this->run_single, this->empty_linestring(), this->empty_multipolygon(), {});
+};
+
+TYPED_TEST(PairwiseLinestringPolygonDistanceTest, MultiToSingleEmpty)
+{
+  CUSPATIAL_RUN_TEST(this->run_single, this->empty_multilinestring(), this->empty_polygon(), {});
+};
+
+TYPED_TEST(PairwiseLinestringPolygonDistanceTest, MultiToMultiEmpty)
+{
+  CUSPATIAL_RUN_TEST(
+    this->run_single, this->empty_multilinestring(), this->empty_multipolygon(), {});
+};
+
+TYPED_TEST(PairwiseLinestringPolygonDistanceTest, SingleToSingleOnePair)
```

Comment:
Generally we test real data cases in the header-only API tests only. To avoid redundant tests, the column-based API tests just test error cases and empty input cases.

If there are data cases tested here that are not in the header-only API tests, they should probably move there and any duplicate data tests removed from here.

Reply:
These are duplicates. Probably remnants from my tests regarding some confounding bug I found in Python, attempting to trace all the way down.

Reply:
Removed.